app/models/user.rb

```diff
@@ -53,34 +66,43 @@ def send_activation_email
     UserMailer.account_activation(self).deliver_now
   end
 
-  class << self
-    # return the hash digest of the given string
-    def digest string
-      cost = BCrypt::Engine::MIN_COST if ActiveModel::SecurePassword.min_cost
-      cost = BCrypt::Engine.cost unless ActiveModel::SecurePassword.min_cost
-      BCrypt::Password.create(string, cost: cost)
-    end
+  # Create token and save databse attribute reset_digest
+  def create_reset_digest
+    self.reset_token = User.new_token
 
-    # create new token
-    def new_token
-      SecureRandom.urlsafe_base64
-    end
+    update(reset_digest: User.digest(reset_token), reset_send_at: Time.zone.now)
   end
 
-  private
+  # use UserMailer chapter11 send email now
+  def send_password_reset_email
+    UserMailer.password_reset(self).deliver_now
+  end
 
-  # Create token activated email
-  def create_activation_digest
-    self.activation_token = User.new_token
-    self.activation_digest = User.digest(activation_token)
+  # check time after sendMail
+  def password_reset_expired?
+    reset_send_at > 2.hours.ago
   end
 
-  # Save email user lowercase letter
-  def downcase_email
-    email.downcase!
+  def feed
+    following_ids = "SELECT followed_id FROM relationships
+                     WHERE follower_id = :user_id"
+    Micropost.where("user_id IN (#{following_ids})
```

Comment:
`where("user_id IN (#{following_ids})     OR user_id = :user_id", user_id: id)`
viết scope cho code này